src/backend/toolmanager/toolmanager.ts

```diff
@@ -0,0 +1,346 @@
+            throw error
+          })
+        break
+      }
+      case Repositorys.DXVK: {
+        await fetchReleases({
+          url: DXVK_URL,
+          type: 'DXVK',
+          count: count
+        })
+          .then((fetchedReleases: VersionInfo[]) => {
+            releases.push(...fetchedReleases)
+          })
+          .catch((error: Error) => {
+            throw error
+          })
+        break
+      }
+      case Repositorys.DXVK_ASYNC: {
+        await fetchReleases({
+          url: DXVK_ASYNC_URL,
+          type: 'DXVK-Async',
+          count: count
+        })
+          .then((fetchedReleases: VersionInfo[]) => {
+            releases.push(...fetchedReleases)
+          })
+          .catch((error: Error) => {
+            throw error
+          })
+        break
+      }
+      case Repositorys.DXVK_NVAPI: {
+        await fetchReleases({
+          url: DXVK_NVAPI_URL,
+          type: 'DXVK-NVAPI',
+          count: count
+        })
+          .then((fetchedReleases: VersionInfo[]) => {
+            releases.push(...fetchedReleases)
+          })
+          .catch((error: Error) => {
+            throw error
+          })
+        break
+      }
+      case Repositorys.VKD3D: {
+        await fetchReleases({
+          url: VKD3D_URL,
+          type: 'VKD3D',
+          count: count
+        })
+          .then((fetchedReleases: VersionInfo[]) => {
+            releases.push(...fetchedReleases)
+          })
+          .catch((error: Error) => {
+            throw error
+          })
+        break
+      }
+      default: {
+        logWarning(
+          `Unknown and not supported repository key passed! Skip fetch for ${repo}`,
+          {
+            prefix: LogPrefix.ToolManager
+          }
+        )
+        break
+      }
+    }
+  }
+  return releases
+}
+
+interface installProps {
+  versionInfo: VersionInfo
+  installDir: string
+  overwrite?: boolean
+  onProgress?: (state: State, progress?: ProgressInfo) => void
+  abortSignal?: AbortSignal
+}
+
+/**
+ * Installs a given version to the given installation directory.
+ *
+ * @param versionInfo the version to install as {@link VersionInfo}
+ * @param installDir absolute path to installation directory
+ * @param overwrite allow overwriting existing version installation
+ * @defaultValue false
+ * @param onProgress callback to get installation progress
+ * @defaultValue void
+ * @returns * resolves with updated {@link VersionInfo} and the full installation
+ *            directory
+ *          * rejects with an {@link Error}
+ */
+async function installVersion({
+  versionInfo,
+  installDir,
+  overwrite = false,
+  onProgress = () => {
+    return
+  },
+  abortSignal
+}: installProps): Promise<{ versionInfo: VersionInfo; installDir: string }> {
+  /*
+   * VARIABLE DECLARATION
+   */
+
+  const tarFile =
+    installDir + '/' + versionInfo.download.split('/').slice(-1)[0]
+  const installSubDir = installDir + '/' + versionInfo.version
```

Comment:
is `/` cross-platform compatible? I know I had issues in the past and I always fallback to something like the `path.join` from the path module that ensures compatibility

I guess it's fine though since I don't think we show the tool manager on windows anyway

Reply:
Good point. Actually this is the old code from heroic-wine-downloader and i don't want to change things in this PR.
We should do it in small PR's after this is merged.